### PR TITLE
Additional filesystem notes (CACHEDIR / localstatedir)

### LINKED ISSUE #85: Add details of SINGULARITY_CACHEDIR requirements
Detail space needed, atomic rename for concurrency.

### PR DESCRIPTION
**Pull request by dctrud**
Add notes about SINGULARITY_CACHEDIR requirements (Fixes #85)

Add a bit more information about `--localstatedir`

### CODE REVIEW

File: installation.rst [hunk 1]
```diff
@@ -56,7 +56,19 @@ parallel / network filesystems. In general:
...
+
+   Set the ``--localstatedir`` location by by providing
+   ``--localstatedir my/dir`` as an option when you configure your
+   Singularity build with ``./mconfig``.
 
+   Disk usage at the ``--localstatedir`` location is neglible
+   (<1MiB). The directory is used as a location to mount the container
+   root filesystem, overlays, bind mounts etc. that construct the
+   runtime view of a container. You will not see these mounts from a
+   host shell, as they are made in a separate mount namespace.
```
Comment (vsoch) [inline installation.rst:69]: Maybe here link to the docs you wrote about the different fileystems that would need this done?

File: installation.rst [hunk 2]
```diff
@@ -101,7 +113,46 @@ uid/gid mapping in a user namespace. Because the fileserver is not
...
+Singularity cache / atomic rename
+---------------------------------
+
+Singularity will cache SIF container images generated from remote
+sources, and any OCI/docker layers used to create them. The cache is
+created at ``$HOME/.singularity/cache`` by default. The location of
+the cache can be changed by setting the ``SINGULARITY_CACHEDIR``
+environment variable.
+
+The directory used for ``SINGULARITY_CACHEDIR`` should be:
```
Comment (vsoch) [inline installation.rst:126]: Maybe mention how they are named? For example when you pull (or grab an image from the cache) it's based on the uri, but (at least I think I remember) the cache stores the images named based on hash.

File: installation.rst (hunk 1 above)
Comment (dctrud) [reply installation.rst:69]: This is really independent of the other bits about filesystems. We really *always* want ``--localstatedir`` to be on a local fs to be certain of no weird issues.

The other filesystem limitations are for specific things, like overlay, fakeroot... when certain things are on them - and it's okay to work around those.

File: installation.rst (hunk 2 above)
Comment (dctrud) [reply installation.rst:126]: It has actually been rewritten in 3.6 so names are not involved at all in the cache... every cache item is 'named' purely by hash now, and this is talked about in a userdocs PR here: https://github.com/hpcng/singularity-userdocs/pull/330

**Comment by vsoch** [comment]
For the PR I linked, the issue they ran into was a `singularity <exec/run>` would always require network connectivity, even if the image is in the cache (because the most recent hash needs to be looked up from the manifest to identify the right sif). Is there anyway around this, or a way singularity could support a mode of finding images in the cache without querying for a manifest? That's related to the naming issue - because we need the hash to know the container.

File: installation.rst (hunk 2 above)
Comment (vsoch) [reply installation.rst:126]: Exactly - that is the issue I mentioned below. You can have an image in the cache, but you can't identify it without network connectivity.

**Pull request merged**